dtfs-central-api/src/helpers/calculate-fixed-fee-from-days-remaining.test.ts

```diff
@@ -0,0 +1,30 @@
+import { addDays, startOfMonth, subMonths } from 'date-fns';
+import { calculateFixedFeeFromDaysRemaining } from './calculate-fixed-fee-from-days-remaining';
+import { getNumberOfDaysInCoverPeriod } from '../services/state-machines/utilisation-report/event-handlers/helpers/calculate-fixed-fee';
+
+describe('calculateFixedFeeFromDaysRemaining', () => {
+  it('should return the product of the utilisation, interest percentage (divided by 100), bank admin fee percentage (fixed at 0.9) and number of days remaining in the cover period divided by the day count basis and rounds to 2 decimal places', () => {
+    // Arrange
+    const utilisation = 100000;
+    const interestPercentage = 5;
+    const dayCountBasis = 365;
+
+    const coverStartDateBeforeReportPeriod = subMonths(new Date(), 1);
+
+    // Number of days remaining = 432
+    const coverEndDate = addDays(startOfMonth(new Date()), 432);
+
+    const numberOfDaysRemainingInCoverPeriod = getNumberOfDaysInCoverPeriod(coverStartDateBeforeReportPeriod, coverEndDate);
+
+    // Act
+    const result = calculateFixedFeeFromDaysRemaining({
+      utilisation,
+      numberOfDaysRemainingInCoverPeriod,
+      interestPercentage,
+      dayCountBasis,
+    });
+
+    // Assert
+    expect(result).toBe(5671.23); // 100000 * (5 / 100) * 0.9 * 4603 / 365 = 5,671.2328767... = 5671.23 (ROUNDED)
```

Comment:
- `.toEqual`
- `expected`

Can we move the comment into it's own line - also wondering if we should use some variables instead of the numbers, to understand the calculation you have to go back and forth between the variables and the numbers

Instead of having a hardcoded number, can we/should we use methods like in the actual function?

Reply:
We shouldn't use methods like in the actual function because that is duplicating the implementation and means the test would be verifying we can write the same code twice, rather than giving us confidence that the calculation is correct.

Reply:
changed to toEqual and moved to its own line

Reply:
Pros and cons - currently the test assertion is effectively manually / hard coding the result of the implemention which uses `Big`. In which case why do we need to use `Big`?

I am happy for the current approach to be merged since we probably have this in all other Big related unit tests.

Reply:
This is consistent with already existing tests and this was moved across.  We should update this later though

Reply:
It's stay like this for now so all good etc., but I'd really like to not update this. Big is very important for catching the complete edge cases of how javascript sucks at doing calculations, and having the tiny chance of being 1p off is worth using Big for. If we have a use case of numbers that we know won't cause this edge effect all we are doing here is making sure that no changes happen impact the result of our calculations without us knowing. If we start using methods then we might run the risk of something changing in the calculation but the test not knowing.

I think all I'm getting to is why would we not want to have (at least one) test with this behaviour?

Reply:
Makes sense @ChristianMcc just I think that such tests read very messily. Should be able to read a line and understand what it's doing, this is not the case here:

```ts
expect(result).toBe(5671.23); // 100000 * (5 / 100) * 0.9 * 4603 / 365 = 5,671.2328767... = 5671.23 (ROUNDED)
```

I would expect to have e.g `expect(result).toEqual(expected)`. Or even better something like this:

```ts
const someSubCalculation = varA + varB;
const someOtherNumber = varC * varD;

const expected = someSubCalculation / someOtherNumber + etc
```

This part in particular is messy, hard to read and maintain.

```ts
100000 * (5 / 100) * 0.9 * 4603 / 365 = 5,671.2328767... = 5671.23 (ROUNDED)
```